orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationStreamingHandler.java

```diff
@@ -0,0 +1,55 @@
+package com.sap.ai.sdk.orchestration;
+
+import static com.sap.ai.sdk.orchestration.OrchestrationClient.JACKSON;
+import static com.sap.ai.sdk.orchestration.OrchestrationResponseHandler.buildExceptionAndThrow;
+import static com.sap.ai.sdk.orchestration.OrchestrationResponseHandler.parseErrorAndThrow;
+
+import java.io.IOException;
+import java.util.stream.Stream;
+import javax.annotation.Nonnull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.hc.core5.http.ClassicHttpResponse;
+
+@Slf4j
+@RequiredArgsConstructor
+class OrchestrationStreamingHandler<D extends StreamedDelta> {
+
+  @Nonnull private final Class<D> deltaType;
+
+  /**
+   * @param response The response to process
+   * @return A {@link Stream} of a model class instantiated from the response
+   */
+  @SuppressWarnings("PMD.CloseResource") // Stream is closed automatically when consumed
+  @Nonnull
+  Stream<D> handleResponse(@Nonnull final ClassicHttpResponse response)
+      throws OrchestrationClientException {
+    if (response.getCode() >= 300) {
+      buildExceptionAndThrow(response);
+    }
+    return IterableStreamConverter.lines(response.getEntity())
+        // half of the lines are empty newlines, the last line is "data: [DONE]"
+        .peek(line -> log.info("Handler: {}", line))
```

Comment:
(Comment)

I'm surprised we have so many `info` log statements. I feel like we could define a coding guideline when to use what log level.

(no action required.)